### PR TITLE
Refactors objective code to account for there being multiple owners (team objectives)

### PR DESCRIPTION
**Pull request by SteelSlayer**
**I highly recommend that this PR be test merged for a while.**

## What Does This PR Do
Refactors objective code to account for there being multiple owners (team objectives). 
This needs to happen before I can make my antag team framework PR.

Currently, a `check_completion()` proc for an objective might look something like this:
```dm
/datum/objective/my_obj/check_completion()
	if(owner.current.stat == DEAD)
		return FALSE
	return TRUE
```
Under the new system, we need to account for an objective having multiple owners, so we need to use the `get_owners()` proc to grab ALL owners, including the ones from a team.
```dm
/datum/objective/my_obj/check_completion()
	for(var/datum/mind/M in get_owners())
		if(M.current.stat == DEAD)
			return FALSE
	return TRUE
```
This doesn't apply to just `check_completion`. This applies to anywhere where you're checking some condition on the objective owner(s). Also note that it's fine to use `owner` over `get_owners()` if we're certain that objective will never belong to a team.

Additionally cleans up some things:
* Removes `/datum/objective/download`. This is barebones and doesn't even have code that allows the owner to succeed.
* Removes `/datum/objective/capture`, for the same reason.
* Removes `/datum/objective/steal/exchange`. These objectives haven't been used for years.

## Testing
Loaded up a test server and ran `check_completion()` on all objectives that I could. Still, testing each objective completion in depth is annoying and time consuming, so I still want a TM for this PR.

## Changelog
Should have no player-facing changes.

### CODE REVIEW

**Comment by GDNgit** [comment]
this makes me very happy :>

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -290,47 +308,39 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
...
-
-	if(istype(location, /turf/simulated/floor/mineral/plastitanium/red/brig)) // Fails traitors if they are in the shuttle brig -- Polymorph
-		return 0
-
-	if(location.onCentcom() || location.onSyndieBase())
-		return 1
+		return TRUE
+	var/list/owners = get_owners()
+	for(var/datum/mind/M in owners)
+		if(M.current?.stat == DEAD || issilicon(M.current) || isbrain(M.current))
```
Comment (S34NW) [inline code/game/gamemodes/objective.dm:335]: If there is no `M.current` this if won't evaluate and will result in an eventual return of TRUE in certain cases. That seems unintentional.

File: code/game/gamemodes/objective.dm [hunk 2]
```diff
@@ -381,11 +395,12 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 	needs_target = FALSE
 
 /datum/objective/survive/check_completion()
-	if(!owner.current || owner.current.stat == DEAD || isbrain(owner.current))
-		return 0		//Brains no longer win survive objectives. --NEO
-	if(issilicon(owner.current) && !owner.is_original_mob(owner.current))
-		return 0
-	return 1
+	for(var/datum/mind/M in get_owners())
+		if(M.current?.stat == DEAD || isbrain(M.current))
```
Comment (S34NW) [inline code/game/gamemodes/objective.dm:399]: same here as above with `M.current?.stat == DEAD`

File: code/game/gamemodes/objective.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/game/gamemodes/objective.dm:335]: You're right, I'll need to still check `!M.current || M.current.stat == DEAD`

File: code/game/gamemodes/miniantags/revenant/revenant.dm [hunk 3]
```diff
@@ -320,15 +320,15 @@
...
-		return 0
-	var/mob/living/simple_animal/revenant/R = owner.current
-	if(!R || R.stat == DEAD)
-		return 0
-	var/essence_stolen  = R.essence_accumulated
-	if(essence_stolen  < targetAmount)
-		return 0
-	return 1
+	var/total_essence = 0
+	for(var/datum/mind/M in get_owners())
```
[the remark this answers is not in the document]
Comment (SteelSlayer) [reply code/game/gamemodes/miniantags/revenant/revenant.dm:324]: I don't think we'll ever have team revenents, but figured I should do it just in case.

**Review by S34NW — APPROVED** [review]
From what I can see this looks alright now, but for sure it needs a TM to be certain as these are round-critical changes

**Review by S34NW — CHANGES_REQUESTED** [review]
Updating my approval as this needs to be TMd

**Comment by S34NW** [comment]
Any runtimes or issues relating to this? I note there has been an uptick in free objectives

**Comment by SteelSlayer** [comment]
The free objective thing *was* caused by this PR. Fixed in the latest commit by removing a single `!`.

**Comment by SteelSlayer** [comment]
Fixed a bug with where if you have the escape objective, you're off station, and the AI doomsdays/station is nuked, the objective would fail.

File: code/game/gamemodes/objective.dm [hunk 4]
```diff
@@ -72,9 +94,9 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
...
   */
 /datum/objective/proc/on_target_cryo()
-	if(owner?.current)
-		to_chat(owner.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
-		SEND_SOUND(owner.current, sound('sound/ambience/alarm4.ogg'))
+	for(var/datum/mind/M in get_owners())
+		to_chat(M.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
+		SEND_SOUND(M.current, sound('sound/ambience/alarm4.ogg'))
 	target = null
 	INVOKE_ASYNC(src, .proc/post_target_cryo)
```
Comment (hal9000PR) [inline code/game/gamemodes/objective.dm:101]: to save doing `get_owners` twice couldnt you cashe it and pass it to `post_target_cryo` not sure if its called anywhere else

File: code/game/gamemodes/objective.dm [hunk 5]
```diff
@@ -351,28 +365,32 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
...
 	if(!target_real_name)
-		return 1
+		return TRUE
 	if(!ishuman(owner.current))
-		return 0
+		return FALSE
 	var/mob/living/carbon/human/H = owner.current
 	if(..())
 		if(H.dna.real_name == target_real_name)
 			if(H.get_id_name()== target_real_name)
```
Comment (hal9000PR) [inline code/game/gamemodes/objective.dm:378]:
```suggestion
			if(H.get_id_name() == target_real_name)
```
nitpick

File: code/game/gamemodes/objective.dm [hunk 6]
```diff
@@ -468,65 +486,15 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
...
-
+	for(var/datum/mind/M in get_owners())
+		var/mob/living/carbon/human/mob = M.current
+		var/where = mob.equip_in_one_of_slots(I, slots)
+		if(where)
+			to_chat(mob, "<br><br><span class='info'>In your [where] is a box containing <b>items and instructions</b> to help you with your steal objective.</span><br>")
+		else
+			to_chat(mob, "<span class='userdanger'>Unfortunately, you weren't able to get a stealing kit. This is very bad and you should adminhelp immediately (press F1).</span>")
+			message_admins("[ADMIN_LOOKUPFLW(mob)] Failed to spawn with their [item_path] theft kit.")
+			qdel(I)
```
Comment (hal9000PR) [inline code/game/gamemodes/objective.dm:497]: please dont use `mob` as a var. I know it wasnt you.

**Review by hal9000PR — CHANGES_REQUESTED** [review]
some minor nits and should be gtg

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**